doordeck-sdk/src/commonMain/kotlin/com/doordeck/multiplatform/sdk/internal/api/AccountClient.kt

```diff
@@ -52,10 +56,23 @@ internal open class AccountClient(
      * @see <a href="https://developer.doordeck.com/docs/#register-ephemeral-key">API Doc</a>
      */
     suspend fun registerEphemeralKeyRequest(publicKey: ByteArray): RegisterEphemeralKeyResponse {
+        return registerEphemeralKey(publicKey)
+    }
+
+    suspend fun registerEphemeralKeyWithContextRequest(): RegisterEphemeralKeyResponse {
+        val publicKey =  contextManager.getPublicKey()
+            ?: throw MissingContextFieldException("Key pair is missing")
+        return registerEphemeralKey(publicKey)
+    }
+
+    private suspend fun registerEphemeralKey(publicKey: ByteArray): RegisterEphemeralKeyResponse {
         val publicKeyEncoded =  publicKey.encodeByteArrayToBase64()
-        return httpClient.post(Paths.getRegisterEphemeralKeyPath()) {
+        return httpClient.post<RegisterEphemeralKeyResponse>(Paths.getRegisterEphemeralKeyPath()) {
             addRequestHeaders()
             setBody(RegisterEphemeralKeyRequest(publicKeyEncoded))
+        }.also {
+            contextManager.setUserId(it.userId)
+            contextManager.setCertificateChain(it.certificateChain)
```

Comment:
From now on, we will update the context values whenever possible. Previously, this had to be done manually by the developer. Please note that this only updates the variable; it does not persist it.